TemporaryProjects.14.0/StartPageExtender.cs

```diff
@@ -0,0 +1,195 @@
+
+        // You can test this in the current Visual Studio instance like this:
+        // https://github.com/jcansdale/TestDriven.Net-Issues/wiki/Test-With...VS-SDK
+        [STAThread]
+        public static void Initialize(IVsUIShell7 vsUIShell, DTE dte)
+        {
+            ReadviseWindowFrameEvents(vsUIShell, dte);
+        }
+
+        private static void ReadviseWindowFrameEvents(IVsUIShell7 vsUIShell, DTE dte)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (Cookie is uint cookie)
+            {
+                // Ensure that we only have one event sink installed for the app domain if initialize is called
+                // multiple times while testing.
+                vsUIShell.UnadviseWindowFrameEvents(cookie);
+            }
+
+            Cookie = vsUIShell.AdviseWindowFrameEvents(new StartPageExtender(dte, (IVsUIShell)vsUIShell));
+        }
+
+        // Store cookie in app domain wide variable because assembly might be loaded multiple times while testing.
+        static uint? Cookie
+        {
+            get => (uint?)AppDomain.CurrentDomain.GetData(CookieName);
+            set => AppDomain.CurrentDomain.SetData(CookieName, value);
+        }
+
+        private StartPageExtender(DTE dte, IVsUIShell vsUIShell)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            this.dte = dte;
+
+            if (ErrorHandler.Succeeded(vsUIShell.FindToolWindow(
+                    (uint)__VSFINDTOOLWIN.FTW_fFrameOnly, VSConstants.StandardToolWindows.StartPage, out var windowFrame)) &&
+                windowFrame != null)
+            {
+                TrackFrame(windowFrame);
+            }
+        }
+
+        public void OnFrameCreated(IVsWindowFrame frame)
+        {
+            // When the frame for the start page is created, it is empty and has no content. Not only that, but
+            // the content of the frame seems to be recreated every time the tab is activated, so we have to
+            // extend its content in OnFrameIsOnScreenChanged. This may happen multiple times with the same frame. If we only
+            // did this once, our button would be gone if the user switched to another tab and then back to the start page.
+        }
+
+        public void OnFrameDestroyed(IVsWindowFrame frame)
+        {
+        }
+
+        public void OnFrameIsVisibleChanged(IVsWindowFrame frame, bool newIsVisible)
+        {
+        }
+
+        public void OnFrameIsOnScreenChanged(IVsWindowFrame frame, bool newIsOnScreen)
+        {
+            // If you try to make any improvements here, please make sure not only that the button is visible
+            // when the start page is opened, but also:
+            // 1. When the user switches to another tab and back to the start page
+            // 2. When the start page is undocked by dragging the tab
+            // 3. When the start page is docked again (this currently *doesn't* work)
```

Comment:
@jcansdale Yes, you can merge if you feel the current state is good enough and it's ok to have this bug

Reply:
Thanks for all your work on this!

I think undocking and re-docking will be very much an edge case. If it didn't work when stwitching tabs, that would be confusing. Its current state is certainly good enough! 🎉

Reply:
Filed #8 as a followup issue